pkg/controller/shoot/shoot_care_control.go

```diff
@@ -221,24 +207,41 @@ func garbageCollection(botanist *botanistpkg.Botanist) {
 // It receives a Garden object <garden> which stores the Shoot object.
 // The current Health check verifies that the control plane running in the Seed cluster is healthy, every
 // node is ready and that all system components (pods running kube-system) are healthy.
-func healthCheck(botanist *botanistpkg.Botanist, cloudBotanist cloudbotanist.CloudBotanist, conditionControlPlaneHealthy, conditionEveryNodeReady, conditionSystemComponentsHealthy *gardenv1beta1.Condition) (*gardenv1beta1.Condition, *gardenv1beta1.Condition, *gardenv1beta1.Condition) {
-	var wg sync.WaitGroup
+func healthCheck(botanist *botanistpkg.Botanist, controlPlaneCondition, clusterNodesCondition, systemComponentsCondition *gardenv1beta1.Condition) (*gardenv1beta1.Condition, *gardenv1beta1.Condition, *gardenv1beta1.Condition) {
+	var (
+		wg                               sync.WaitGroup
+		updatedControlPlaneCondition     *gardenv1beta1.Condition
+		updatedClusterNodesCondition     *gardenv1beta1.Condition
+		updatedSystemComponentsCondition *gardenv1beta1.Condition
+	)
 
 	wg.Add(3)
 	go func() {
 		defer wg.Done()
-		conditionControlPlaneHealthy = botanist.CheckConditionControlPlaneHealthy(conditionControlPlaneHealthy)
+		var err error
+		updatedControlPlaneCondition, err = botanist.CheckControlPlane(controlPlaneCondition)
```

Comment:
Why explicitly declaring
```golang
var err error
```
? What about
```suggestion
		updatedControlPlaneCondition, err := botanist.CheckControlPlane(controlPlaneCondition)
```

(Same for other occurrences here)

Reply:
If you not explicitly declare `err error` , the assignment to `updatedControlPlaneCondition` and others would only be visible in the goroutine but not outside for the final return.

Reply:
OK, I missed that one. Thanks.